### DISCUSSION TITLE
Where can I find the documentation on previous versions?

### DISCUSSION THREAD

**Discussion by EcksDy** [nrwl/nx#19101] [General]
We're currently on nx version 14.1.7, and do not intend to upgrade for the foreseeable future.

I wasn't able to find documentation for that version, and a lot has changed since then. Trying to look at commits and approximate their time to the time of the version is tedious.

Is there something I'm missing? Is there a navigation similar to nestjs where it links to previous versions?
![image](https://github.com/nrwl/nx/assets/994443/c0774d8d-8cd0-4a79-b405-5ce5ccc5ea9d)

**Comment by ptu14** [comment]
I have the same case, migrations do not take place immediately, and we want to add new features and then migrate them

**Comment by LayZeeDK** [comment]
Definitely a feature that would be great. Right now, your only option would be to navigate to the branch matching the version you use and run the documentation website locally. However, I tried this for older versions (13.x I think) without any success so your mileage might vary.

**Comment by zaldih** [comment]
+1 to this. A drop-down to select the version of the documentation would be great.

**Comment by kahzew** [comment]
The google wayback machine seems to have at least crawled v17:

https://web.archive.org/web/20250304145340/https://17.nx.dev/getting-started/intro

I haven't tried prior version, not sure if the URL remained the same between them.

**Reply by india-e** [reply]
as far as I can tell only the landing page is available there. but if you select the tag for the version you want in github you can find the old docs source at least e.g. [17.3.2](https://github.com/nrwl/nx/blob/17.3.2/docs/generated/cli/run-many.md)

**Reply by LayZeeDK** [reply]
`https://<version>.nx.dev` works but older versions like `https://17.nx.dev` are removed over time. Currently, the latest major Nx version is 22 while 19 is the oldest version of the documentation to be hosted at https://19.nx.dev.
